maven-plugin/src/test/java/io/confluent/kafka/schemaregistry/maven/derive/schema/DeriveJsonSchemaTest.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2022 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.kafka.schemaregistry.maven.derive.schema;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import io.confluent.kafka.schemaregistry.json.JsonSchema;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Optional;
+
+import static org.junit.Assert.assertEquals;
+
+public class DeriveJsonSchemaTest {
+
+  private final ObjectMapper mapper = new ObjectMapper();
+  DeriveJsonSchema derive = new DeriveJsonSchema();
+
+  public void generateSchemaAndCheckPrimitive(String message, String expectedSchema)
+      throws JsonProcessingException {
+    Optional<ObjectNode> primitiveSchema = derive.getPrimitiveSchema(mapper.readTree(message));
```

Comment:
may be I am confused. DeriveJsonSchema does not have a `getPrimitiveSchema` implementation so shouldn't this test be in DeriveSchemaTest?

Reply:
`getPrimitiveSchema` uses `classToDataType`.  `classToDataType` will be different for each format and hence I have placed it in `DeriveJsonSchemaTest`